### ISSUE TITLE
Complain if bash-like string appending is used in POSIX scripts

### ISSUE THREAD

**Issue by rugk** [koalaman/shellcheck#820]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
# below fails in POSIX
s=test1
s+="test2"

# works below
s=test1
s=$s"test2"
s="${s}test2"
```

#### Here's what shellcheck currently says:

No issues

#### Here's what I wanted or expected to see:

`+=` is not available as a POSIX shell feature, only available in Bash (e.g.)

See also https://stackoverflow.com/questions/2250131/how-do-you-append-to-an-already-existing-string#answer-2250199

**Comment by kurahaupo** [comment]
There does not appear to be a += in the example

**Comment by rugk** [comment]
You're right, I updated the issue report.

**Comment by koalaman** [comment]
Definitely a glaring omission in the bashism checks. Fixed in 1863f2f, thanks!
